utils/src/userInput/CopilotUserInput.ts

```diff
@@ -0,0 +1,123 @@
+import { workspace } from 'vscode';
+import * as types from '../../index';
+import { createPrimaryPromptForInputBox, createPrimaryPromptForWarningMessage, createPrimaryPromptForWorkspaceFolderPick, createPrimaryPromptToGetPickManyQuickPickInput, createPrimaryPromptToGetSingleQuickPickInput, doCopilotInteraction } from '../copilot/copilot';
+import { InvalidInputError } from '../errors';
+
+export class CopilotUserInput implements types.IAzureUserInput {
+    private readonly _vscode: typeof vscodeTypes;
+    private readonly _onDidFinishPromptEmitter: vscodeTypes.EventEmitter<types.PromptResult>;
+    private readonly _relevantContext: string | undefined;
+
+    constructor(vscode: typeof vscodeTypes, relevantContext?: string) {
+        this._vscode = vscode;
+        this._onDidFinishPromptEmitter = new this._vscode.EventEmitter<types.PromptResult>();
+        this._relevantContext = relevantContext;
+    }
+
+    public async showWarningMessage<T extends vscodeTypes.MessageItem>(message: string, ...items: T[]): Promise<T> {
+        const primaryPrompt: string = createPrimaryPromptForWarningMessage(message, items);
+        const response = await doCopilotInteraction(primaryPrompt)
+
+        const pick = items.find(
+            item => {
+                return item.title === response;
+            }
+        );
+
+        if (!pick) {
+            throw new InvalidInputError();
+        }
+
+        this._onDidFinishPromptEmitter.fire({ value: pick });
+        return pick;
+    }
+
+    public showOpenDialog(): Promise<vscodeTypes.Uri[]> {
+        // Throw this back to the user
+        throw new InvalidInputError();
+    }
+
+    public async showWorkspaceFolderPick(_options: types.AzExtWorkspaceFolderPickOptions,): Promise<vscodeTypes.WorkspaceFolder> {
+        const primaryPrompt: string = createPrimaryPromptForWorkspaceFolderPick(workspace.workspaceFolders, this._relevantContext);
+        const response = await doCopilotInteraction(primaryPrompt)
+        const pick = (workspace.workspaceFolders || []).find(folder => {
+            return folder.name === response;
+        });
+
+        if (!pick) {
+            throw new InvalidInputError();
+        }
+
+        this._onDidFinishPromptEmitter.fire({ value: pick });
+        return pick;
+    }
+
+    public async showInputBox(options: vscodeTypes.InputBoxOptions): Promise<string> {
+        if (options.prompt) {
+            try {
+                const primaryPrompt: string = createPrimaryPromptForInputBox(options.prompt, this._relevantContext);
+                const response = await doCopilotInteraction(primaryPrompt)
+                const jsonResponse: string = JSON.parse(response) as string;
+                this._onDidFinishPromptEmitter.fire({ value: jsonResponse });
+                return jsonResponse;
+            } catch {
+                // if copilot is unable to provide a response, fall back to the default value if it exists
+                if (options.value) {
```

Comment:
Shouldn't this be outside of `if(options.prompt)`?  If there's no prompt, shouldn't we default to `options.value`?